test/net/sourceforge/kolmafia/textui/javascript/AshInteropTest.java

```diff
@@ -71,4 +72,25 @@ void getClassFromId() {
     String retS = ret.toString();
     assertEquals("Seal Clubber", retS);
   }
+
+  @Test
+  void nowToIntHandlesLongsCorrectly() {
+    var js = new JavascriptRuntime("nowToInt()");
+    assertNotNull(js, "JavascriptRuntime returned as null.");
+    Value ret = js.execute(null, null, true);
+    assertNotNull(ret, "Javascript execute returns null instead of a result to be tested.");
+    long retVal = Long.parseLong(ret.toString());
+    assertTrue(retVal >= 0, "Now should not be negative.");
+
+    long javaNow = Instant.now().toEpochMilli();
+    long diff = Math.abs(javaNow - retVal);
+    // there's no asertCloseTo() ...
```

Comment:
There is `assertThat(x, closeTo(y, z));`, though.

If you search the codebase for `closeTo` you'll find some tests using it.

Reply:
I do see that.  But the only implementation we have is for doubles, so I'd have to worry about that conversion.

And when I look at the implementation from another project of `closeToInteger()`, it's the same function I have (abs (val-val) > margin).

Do you think I should use closeTo()?  It's very readable, that's for sure.

Reply:
oh right, I didn't see what you were testing.

Can you use `dateToTimestamp` with a fixed date instead? That would obviate the need for a "close to".

Reply:
that wouldn't really test nowToInt(), though.  I'm open to changing it, just not sure it's viable.

Reply:
It wouldn't, but your change is "timestamps that are too high are correctly converted to longs, not negative numbers" -- nothing specific about `nowToInt()`.

e.g. at the moment we have
> > js print(dateToTimestamp("yyyy-MM-dd", "2023-03-03"));
>
> -1530612736

but this should not be negative.

Reply:
well, it specifically came up in the context of nowToInt, but you're right.  The fix is larger in scope than dates

> > js print(dateToTimestamp("yyyy-MM-dd", "2023-03-03"));

1677823200000
Returned: null

Reply:
done.